src/Dfe.EarlyYearsQualification.Content/Services/ContentfulContentService.cs

```diff
@@ -140,6 +141,19 @@ public ContentfulContentService(IContentfulClient contentfulClient, ILogger<Cont
         return await GetEntryById<QuestionPage>(entryId);
     }
 
+    public async Task<PhaseBanner?> GetPhaseBannerContent()
+    {
+      var phaseBannerEntities = await GetEntriesByType<PhaseBanner>();
+      if (phaseBannerEntities is null || !phaseBannerEntities.Any())
+      {
+          _logger.LogWarning("No phase banner entry returned");
+          return default;
+      }
+      var phaseBannerContent = phaseBannerEntities.First();
+      phaseBannerContent.ContentHtml = await GetPhaseBannerRenderer().ToHtml(phaseBannerContent.Content);

```

Comment:
suggestion: HTML-rendering logic should not be the responsibility of this service?

Single responsibility principle: I'd like to suggest that this service should be responsible only for getting the content from Contentful? I think the rendering would be better done later and elsewhere. With this current pattern, it's possible for the C# code to change the value of the `ContentHtml` on the object after it's been returned from this service. Wouldn't it be better for the `PhaseBanner` object not to have a `ContentHtml` property at all, and for the renderer to calculate it at just the point it is needed?

I realise this is a discussion point, and if we all agree on this we'd want to refactor the way this is done elsewhere too. I'd be happy to do that myself in another branch/PR... :)

Reply:
We have a chat about this later

Reply:
Yes, this is a topic for its own branch & PR